src/Hedgehog.Xunit/Attributes.fs

```diff
@@ -100,3 +100,26 @@ type RecheckAttribute(recheckData) =
   let _recheckData : string = recheckData
 
   member internal _.GetRecheckData = _recheckData
+
+/// Set a Generator for a parameter of a test annotated with `Property`
+///
+/// Example usage:
+///
+/// ```
+///
+/// type ConstantInt(i: int) =
+///   inherit ParameterGeneraterBaseType<int>()
+///   override _.Generator = Gen.constant i
+///
+/// [<Property>]
+/// let ``is always 2`` ([<ConstantInt(2)>] i) =
+///   Assert.StrictEqual(2, i)
+///
+/// ```
+[<AbstractClass>]
+[<AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false)>]
+type ParameterGeneraterBaseType<'a>() =
```

Comment:
Okay, bikeshedding time - how do we feel about dropping `BaseType`? I feel like it isn't particularly helpful and I like terse naming.

Also, should this be `ArgumentGenerator`? It _generates arguments_... for the given parameter. Yes the attribute is on a parameter, but I thinks names should reflect what they do, rather than where they're used.

Or, we could sidestep this nonsense entirely and call it `GenAttribute`.

Reply:
This make sense.

Reply:
Wasn't quite sure what you intended, but I changed it to `GenAttribute`. LMK if you feel like it should be something else!